tools/archives/pyvo_integration/astronomical_archives.py

```diff
@@ -1305,6 +1320,17 @@ def collect_resource_keys(urls_data: list) -> list:
                     resource_keys.append(key)
         return resource_keys
 
+    @staticmethod
+    def is_valid_url(url: str) -> bool:
```

Comment:
Maybe you can use here a function from urllib, e.g. urlparse()

Reply:
Thanks, I saw some people using urlparse but apparently it's too permissive in some cases. The one I got is what is or was used to validate url in Django (and I removed local and ip addresses + other protocols).

I will investigate a bit more

Reply:
There is also https://github.com/python-validators/validators which you may consider to use here.
But if regex covers all cases, we can go with it, I think. However, the current regex seems to be a bit wrong, it allows for empty hostname due to `|` in the end of line 1328. 
This could be covered by some kind of test. @bgruening is there an established practice of adding unit-tests (with e.g. pytest) to the tool's script, or it should be incorporated in planemo tests ?

Reply:
unit test will not be run. Planemo test is the only tests that we run automatically. That is, we usually don't want to ship complex code with the wrappers. The usual way is that we depend on packages and more complex scripts should be packaged etc ...

However, we could run unit-tests here on CI for all .py scripts, if that helps.

Reply:
> There is also https://github.com/python-validators/validators which you may consider to use here. But if regex covers all cases, we can go with it, I think. However, the current regex seems to be a bit wrong, it allows for empty hostname due to `|` in the end of line 1328. This could be covered by some kind of test. @bgruening is there an established practice of adding unit-tests (with e.g. pytest) to the tool's script, or it should be incorporated in planemo tests ?

I saw that the validators package seemed to be widely used but it's not built in and you have to install it separately if I'm not mistaken.

Thanks for the find I will correct the regex and then we can see if we keep it or use another solution